data-prepper-core/src/main/java/org/opensearch/dataprepper/plugin/ComponentPluginArgumentsContext.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugin;
+
+import org.opensearch.dataprepper.metrics.PluginMetrics;
+import org.opensearch.dataprepper.model.configuration.PipelineDescription;
+import org.opensearch.dataprepper.model.configuration.PluginSetting;
+import org.opensearch.dataprepper.model.plugin.InvalidPluginDefinitionException;
+import org.opensearch.dataprepper.model.plugin.PluginFactory;
+import org.opensearch.dataprepper.model.event.EventFactory;
+import org.opensearch.dataprepper.model.acknowledgements.AcknowledgementSetManager;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.BeanFactory;
+
+import javax.annotation.Nullable;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Supplier;
+
+/**
+ * An internal class which represents all the data which can be provided
+ * when constructing a new plugin.
+ */
+class ComponentPluginArgumentsContext implements PluginArgumentsContext {
```

Comment:
To help with reviewing - this is the same code from `PluginArgumentsContext` which was changed into an interface in this PR. Because I added a new interface with the same line, the copy is lost.

The only changes here are renaming the class and implementing the new interface.